agent/config/builder.go

```diff
@@ -882,11 +882,15 @@ func (b *Builder) Build() (rt RuntimeConfig, err error) {
 		},
 
 		// Agent
-		AdvertiseAddrLAN:                       advertiseAddrLAN,
-		AdvertiseAddrWAN:                       advertiseAddrWAN,
-		BindAddr:                               bindAddr,
-		Bootstrap:                              b.boolVal(c.Bootstrap),
-		BootstrapExpect:                        b.intVal(c.BootstrapExpect),
+		AdvertiseAddrLAN: advertiseAddrLAN,
+		AdvertiseAddrWAN: advertiseAddrWAN,
+		BindAddr:         bindAddr,
+		Bootstrap:        b.boolVal(c.Bootstrap),
+		BootstrapExpect:  b.intVal(c.BootstrapExpect),
+		Cache: Cache{
+			EntryFetchRateLimit: rate.Limit(b.float64ValWithDefault(c.CacheConfig.EntryFetchRateLimit, 1)),
```

Comment:
The default should be `rate.Inf`, no?

Reply:
Well, that's debatable :-)

As explained in the header of this PR (and in the doc), I have chosen that's the default value is 1qps by default with a burst size of 2.
It means the cache will not do more than 1 qps on servers by default. It sounds reasonable to have such value because otherwise, the cache is doing more harm than dumb calls from RPC APIs.

Of course, that's to HashiCorp to choose the value, but having large values there will break your clients as soon as they use cache. Using it is easy: use Consul Connect. In our experience, greater values than that just hurts the cluster while bringing no benefit. Let this value to Inf is IMHO a mistake.

As proof: all integration tests work with this (including all connect tests), so I think having it a 1qps by default is a good choice, but of course, you choose the value your want.

Reply:
I think it needs to be effectively disabled by default, at least for the initial release. My understanding is that using `rate.Inf` will be the equivalent to disabled. If we were to enable it I think that would be a breaking change. Other users may have smaller clusters and might expect a lot more than 1qps.

In a future major release we could choose to enable it with a different value.

Reply:
Ok, you are the boss, I'll change that tomorrow but I think it is a mistake: the goal of a cache is to improve performance of reads, not to break the cluster :-)

Having sub-second micro-updates for 1h in the cache is not gonna make Consul more efficient, it will break Consul Connect and force people to tune it *AFTER* an outage :-)

Reply:
I think we can probably set a default value here in the future, but I think we want this to go into 1.8.x, so I think we have to be a little more conservative. In 1.9 we could set this to something smaller.

Reply:
@dnephin 3 qps by default? Much more?

Reply:
@dnephin I did put 2 qps instead of 1 by default in https://github.com/hashicorp/consul/pull/8226/commits/593b18b199f87c099bde316d7f11bad9376a4aa8 and now, you just have to change a constant to change the default values (tests are not impacted anymore).

Feel free to tell me the exact value you want (after discussion with your team I imagine)

Reply:
For 1.8.x we should set it to `rate.Inf` and have a discussion if we want to change it for 1.9.

Reply:
Done in b3947c3f218b1e616c05d8f4cb0876cf7a543cd6

Reply:
Set to +Inf